tests/csm/rest/test_capacity_quota.py

```diff
@@ -1927,3 +1927,119 @@ def test_40605(self):
         res = self.csm_obj.get_user_quota("")
         self.log.info("response :  %s", res)
         assert res.status_code == HTTPStatus.NOT_FOUND, "Status code check failed"
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-43857')
+    def test_43857(self):
+        """
+        Verify overwrite scenario for 1 object with check_on_raw=True
+        """
+        test_case_name = cortxlogging.get_frame()
+        test_cfg = self.csm_conf["test_43857"]
+        enabled = test_cfg["enabled"]
+        max_size = test_cfg["max_size"]
+        max_objects = test_cfg["max_objects"]
+        self.log.info("Step 1: Set max_size and max_objects quota for user")
+        payload = self.csm_obj.iam_user_quota_payload(enabled,max_size,max_objects,
+                                                       check_on_raw=True)
+        result, resp = self.csm_obj.verify_get_set_user_quota(self.user_id, payload,
+                                                              verify_response=True)
+        assert result, "Verification for get set user failed."
+        self.log.info("Response : %s", resp)
+        self.log.info("Step 2: Try uploading 1 object(obj-1) of max_size")
+        resp = s3_misc.create_put_objects(self.obj_name, self.bucket,
+                                          self.akey, self.skey, object_size=int(max_size/1024),
+                                          block_size="1K")
+        assert resp, "Put object Failed"
+        self.log.info("Step 3: Try uploading one more object(obj-1) with same name but "
+                      "of different size")
+        try:
+            resp = s3_misc.create_put_objects(self.obj_name, self.bucket,
+                                          self.akey, self.skey,
+                                 object_size=int(max_size/2/1024), block_size="1K")
+        except ClientError as error:
+            self.log.info("Expected exception received %s", error)
+            assert error.response['Error']['Code'] == "QuotaExceeded", \
+                                      "Put operation passed after max size"
+        self.log.info("Step 4: Delete object")
+        assert s3_misc.delete_object(
```

Comment:
Can we take care of clean up in teardown.

Reply:
done